docs/scheduler-perf/scheduler-perf.md

```diff
@@ -0,0 +1,856 @@
+        debug=True
+      databaseInstance: openstack
+      databaseUser: cinder
+      debug:
+        dbSync: false
+      passwordSelectors:
+        database: CinderDatabasePassword
+        service: CinderPassword
+      preserveJobs: false
+      rabbitMqClusterName: rabbitmq
+      secret: osp-secret
+      serviceUser: cinder
+```
+
+The cinder volume service was setup to not start automatically to allow us to
+modify the code and reduce the impact on the node of the volume creation part:
+```bash
+$ oc exec -it cinder-volume-lvm-iscsi-0 -- /bin/bash
+$ vi /usr/lib/python3.9/site-packages/cinder/volume/flows/manager/create_volume.py
+```
+And in the `_create_raw_volume` method changing:
+```python
+   ret = self.driver.create_volume(volume)
+```
+With:
+```python
+   # ret = self.driver.create_volume(volume)
+   ret = {}
+```
+
+Then inside the container the service was started:
+```bash
+$ /usr/local/bin/kolla_set_configs
+$ /usr/local/bin/kolla_start
+```
+
+We leave that terminal alone to let the cinder volume service run.
+
+From the host we now modify the quotas to remove the limit on volumes and
+gigabytes for the tests:
+```bash
+$ oc exec -it openstackclient -- openstack quota set --gigabytes=-1 --volumes=-1 admin
+```
+
+The final preparation step is to prepare the access to RabbitMQ from outside of
+OpenShift.
+
+We add a route to be able to access the RabbitMQ management web server applying
+the following manifest:
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  labels:
+    component: rabbitmq-http
+    public: "true"
+    service: rabbitmq
+  name: rabbitmq-http
+  namespace: openstack
+spec:
+  host: rabbitmq-http-openstack.apps-crc.testing
+  port:
+    targetPort: management
+  to:
+    kind: Service
+    name: rabbitmq
+    weight: 100
+  wildcardPolicy: None
+```
+
+We get the cinder credentials (<user>:<password>) for RabbmitMQ by looking at
+the `/etc/cinder/cinder.conf.d/00-config.conf` file in one of the cinder
+containers and looking at the `transport_url` part, and we store it into an
+environmental variable:
+
+For example for the tests our value was:
+```bash
+RABBIT_CREDS=default_user_hSwFuCitEWMbvW0pmEf:rsu1SKU0K8yFvGEANY0VKiyPpAnK8o_F
+```
+
+### Running the tests
+
+We are going to be using the
+[ApacheBench](https://httpd.apache.org/docs/current/programs/ab.html) to run our
+tests, so we make sure it's installed:
+```bash
+$ sudo dnf -y install httpd-tools
+```
+
+And we finally create the JSON for the request, get the credentials for the
+tests, and run the volume creation 4000 request with a concurrency of 20 using
+the `ab` command:
+```bash
+$ echo '{"volume": {"size": 1, "consistencygroup_id": null, "snapshot_id": null, "name": "40", "description": null, "volume_type": null, "availability_zone": null, "metadata": {}, "imageRef": null, "source_volid": null, "backup_id": null}}' > create-vol.json
+$ token_id=`oc exec -t openstackclient -- openstack token issue -f value -c id`
```

Comment:
oc rsh? though maybe that generates extra garbage when it explains which container within the pod it chooses.